CMakeLists.txt

```diff
@@ -29,6 +65,11 @@ set_property(TARGET objlib PROPERTY POSITION_INDEPENDENT_CODE 1)
 
 add_executable(sampler src/main.cpp $<TARGET_OBJECTS:objlib>)
 include_directories(src/include)
+include_directories("${CMAKE_CURRENT_BINARY_DIR}/src/include")
+
+# configure a header file to pass some of the CMake settings to the source code
+configure_file("${CMAKE_CURRENT_SOURCE_DIR}/src/include/config.h.in"
+               "${CMAKE_CURRENT_BINARY_DIR}/src/include/config.h")
```

Comment:
Maybe move this after line 52. 🤔 It is all about metadata for the new `--version` option.